wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1838,7 +1838,12 @@ impl crate::Adapter<super::Api> for super::Adapter {
             .collect();
         Some(crate::SurfaceCapabilities {
             formats,
-            swap_chain_sizes: caps.min_image_count..=max_image_count,
+            // TODO: Right now we're always trunkating the swap chain
+            // (presumably - we're actually setting the min image count which isn't necessarily the swap chain size)
+            // Instead, we should use extensions when available to wait in present.
+            // See https://github.com/gfx-rs/wgpu/issues/2869
+            maximum_frame_latency: caps.min_image_count.saturating_sub(1)
+                ..=max_image_count.saturating_sub(1),
```

Comment:
this should also have an `.max(1)` on it, since the docstring for `SurfaceCapabilities::maximum_frame_latency` requires at least `1`

Reply:
good catch. I checked the spec and I'd now say my comment is actually wrong. `maximum_frame_latency`==0 (if supported) to my understanding now should mean that gpu/cpu are in sequence (although @cwfitzgerald was skeptical about this before). I'll have to experiment though to confirm